Sources/GRPCProtobuf/Errors/GoogleRPCStatus.swift

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+public import GRPCCore
+internal import SwiftProtobuf
+
+/// An error containing structured details which can be delivered to the client.
+///
+/// This error follows the "richer error model" detailed in the
+/// [gRPC error guide](https://grpc.io/docs/guides/error/) and
+/// [Google AIP-193](https://google.aip.dev/193).
+///
+/// Like an `RPCError`, this error has a `code` and `message`. However it also includes
+/// a list of structured error details which can be propagated to clients. A set of standard
+/// details are provided by ``ErrorDetails``.
+///
+/// As a client you can extract this error from an `RPCError` using `unpackGoogleRPCStatus()`.
+///
+/// > Implementation details:
+/// >
+/// > The error information is transmitted to clients in the trailing metadata of an RPC. It is
+/// > inserted into the metadata keyed by "grpc-status-details-bin". The value of the metadata is
+/// > the serialized bytes of a "google.protobuf.Any" protocol buffers message. The content of which
+/// > is a "google.rpc.Status" protocol buffers message containing the status code, message, and
+/// > details.
+public struct GoogleRPCStatus: Error {
+  /// A code representing the high-level domain of the error.
+  public var code: RPCError.Code
+
+  /// A developer-facing error message, which should be in English.
+  ///
+  /// Any user-facing error message should be localized and sent in the ``details`` field
+  /// or localized by the client.
+  public var message: String
+
+  /// A list of messages that carry the error details.
+  ///
+  /// There is a common set of message types for APIs to use.
```

Comment:
This sentence seems weird to me, doesn't really add any useful information, unless I'm missing something

Reply:
Ah, it's because it came from the status.proto where it can be `google.protobuf.Any`, so it makes sense to callout the common types there. Less so here, I'll update the docs.